packages/migrate/src/commands/DbPull.ts

```diff
@@ -144,22 +146,33 @@ Set composite types introspection depth to 2 levels
       throw new NoSchemaFoundError()
     }
 
-    let schema: string | null = null
+    const schema = await match({ url, schemaPath })
+      .when(
```

Comment:
Could use some small comments about the different cases maybe, so it's easier to read without going to the details?

Reply:
Right, check the new commit!